### DISCUSSION TITLE
Not null variable pattern

### DISCUSSION THREAD

**Discussion by NN---** [dotnet/csharplang#7372] [Language Ideas]
This has been discussed earlier.
I tend to find notation of `not null variable` being easier to read than `{ } variable`.
Would it be possible to rethink this ?

```cs
if (a is not null b)
```

equivalent to
```cs
if (a is { } b)
```

**Comment by FaustVX** [comment]
You can write
```C#
if (a is not null and var b)
```
or
```C#
if (a is <TypeName> b)
```

**Reply by NN---** [reply]
Possible, however the first one is a little bit too verbose, probably will raise a question why you need "and var", and the second requires specifying the exact type.

**Comment by 333fred** [comment]
Duplicate of https://github.com/dotnet/csharplang/discussions/6322.
